### PR TITLE
Add MonoGame v3.6

### PR DESCRIPTION
**Pull request by kinosang**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not already refused in [closed issues].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/caskroom/homebrew-cask/pulls
[closed issues]: https://github.com/caskroom/homebrew-cask/issues?q=is%3Aissue+is%3Aclosed
[the correct repo]: https://github.com/caskroom/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

File: Casks/monogame.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  url "http://www.monogame.net/releases/v#{version}/MonoGame.pkg"
+  name 'MonoGame'
+  homepage 'http://www.monogame.net/'
+
+  pkg 'MonoGame.pkg'
+
+  uninstall_script = "#{staged_path}/uninstall.sh"
+
+  preflight do
+    IO.write uninstall_script, <<~EOS
```
Comment (commitay) [inline Casks/monogame.rb]: Remove this.

File: Casks/monogame.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  uninstall pkgutil: 'com.monogame.*',
+            script:  {
+                       executable: "#{staged_path}/uninstall.sh",
+                       sudo:       true,
+                     },
+            trash:   [
+                       '/Library/Frameworks/MonoGame.framework',
+                       '/usr/local/bin/mgcb',
+                       '/usr/local/bin/monogame-uninstall',
```
Comment (commitay) [inline Casks/monogame.rb]: What does `monogame-uninstall` do?

File: Casks/monogame.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      fi
+    EOS
+  end
+
+  uninstall pkgutil: 'com.monogame.*',
+            script:  {
+                       executable: "#{staged_path}/uninstall.sh",
+                       sudo:       true,
+                     },
+            trash:   [
```
Comment (commitay) [inline Casks/monogame.rb]: Change this to delete

File: Casks/monogame.rb (hunk 1 above)
Comment (kinosang) [reply Casks/monogame.rb]: @commitay why not uninstall the extension of Xamarin Studio?

Comment (commitay) [reply Casks/monogame.rb]: If this needs to be done it should be in a script provided by upstream that we can run, we don't want to have to include the script in the Cask.

**Review by commitay — APPROVED** [review]

File: Casks/monogame.rb (hunk 1 above)
Comment (kinosang) [reply Casks/monogame.rb]: @commitay okay, I'll create an issue or pr to MonoGame.

**Pull request merged**